package.json

```
@@ -33,27 +33,35 @@
   "sideEffects": false,
   "type": "module",
   "main": "index.js",
-  "types": "types/index.d.ts",
+  "types": "index.d.ts",
   "files": [
-    "index.js",
-    "types/index.d.ts"
+    "index.d.ts",
+    "index.js"
   ],
+  "dependencies": {
+    "@types/unist": "^2.0.0"
+  },
   "devDependencies": {
     "@types/mdast": "^3.0.0",
+    "@types/tape": "^4.0.0",
     "c8": "^7.0.0",
-    "dtslint": "^4.0.0",
     "prettier": "^2.0.0",
     "remark-cli": "^9.0.0",
     "remark-preset-wooorm": "^8.0.0",
+    "rimraf": "^3.0.0",
     "tape": "^5.0.0",
+    "tsd": "^0.14.0",
+    "type-coverage": "^2.0.0",
+    "typescript": "^4.0.0",
     "xo": "^0.38.0"
   },
   "scripts": {
+    "prepack": "npm run build && npm run format",
+    "build": "rimraf \"*.d.ts\" && tsc && tsd && type-coverage",
     "format": "remark . -qfo && prettier . -w --loglevel warn && xo --fix",
     "test-api": "node test.js",
     "test-coverage": "c8 --check-coverage --branches 100 --functions 100 --lines 100 --statements 100 --reporter lcov node test.js",
-    "test-types": "dtslint types",
-    "test": "npm run format && npm run test-coverage && npm run test-types"
+    "test": "npm run build && npm run format && npm run test-coverage"
```

Since formatting may break the build and tests, I think formatting should be run first.

```suggestion
    "test": "npm run format && npm run build && npm run test-coverage"
```

---

Reason for doing format after build is that the generated/published types aren't pretty. Running prettier and xo over them ensures they are improved

---

> Running prettier and xo over them ensures they are improved

I personally agree, but this is really subjective. One of the [goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals) of TypeScript is to output clean code. I imagine this applies to the type definitions as well.

> 4. Emit clean, idiomatic, recognizable JavaScript code.

Anyway, these changes have my approval, but personally I think switching them is better, because it may prevent breakage.

---

also experienced it making debugging quite much harder, because the files don’t match the project style if they‘re not formatted. So when inspecting them you first have to dig through tons of formatting issues.

I’d quite want to keep them the order they are, and I’m not 100% on the problems you’re foreseeing by keeping them that way?